hat/app/org/hatdex/hat/api/controllers/Authentication.scala

```diff
@@ -223,4 +225,100 @@ class Authentication @Inject() (
         Future.successful(Unauthorized(Json.toJson(ErrorMessage("Invalid Token", "Token does not exist"))))
     }
   }
+
+  /*
+    BEGIN: hat-claim
+    Entire Section below is on HAT Claim.
+    TODO: Refactor to New File?
+   */
+  /**
+   * Sends an email to the owner with a link to claim the hat
+   */
+  def claim(applicationId: String): Action[AnyContent] = SecuredAction(WithRole(Owner())).async { implicit request =>
+    val email = request.dynamicEnvironment.ownerEmail
+    val response = Ok(Json.toJson(SuccessResponse("You will shortly receive an email with claim instructions")))
+
+    applicationsService.applicationStatus().flatMap { applications =>
+      applications.find(_.application.id.equals(applicationId)) match {
+        case Some(app) =>
+          usersService.listUsers.map(_.find(_.roles.contains(Owner()))).map {
+            case Some(_) =>
+              val token = MailTokenUser(email, isSignUp = false)
```

Comment:
Token generation logic already supports isSignUp boolean. Currently, this flag is not really utilised in the project, so we can use it to indicate claim token (isSignUp: true).
Then, if it's a first call to this endpoint, the token gets created and email sent. 
If it's second or any subsequent call AND the token is not consumed, then the email should be simply resent. 
If the endpoint gets called, token already exists and it's also consumed, then error response should be returned.

Also, the default token expiration time is set to 24h. It should remain the same for the password reset, but should be change to 7 days for the claim token. Please check.

Reply:
1. I know what you mean. But the claim flow don't work this way.
2. Done - Token expiry set to 24h.